webhook/resources/volume/mutator.go

```diff
@@ -97,31 +96,25 @@ func (v *volumeMutator) Create(request *admission.Request, newObj runtime.Object
 		patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/dataLocality", "value": "%s"}`, defaultDataLocality))
 	}
 
-	log.Infof("[volumeMutator] volume.Spec.AccessMode: %v", volume.Spec.AccessMode)
 	if string(volume.Spec.AccessMode) == "" {
 		accessModeFromBackup := longhorn.AccessModeReadWriteOnce
-
 		if volume.Spec.FromBackup != "" {
-			_, bvName, _, err := backupstore.DecodeBackupURL(volume.Spec.FromBackup)
+			bName, _, _, err := backupstore.DecodeBackupURL(volume.Spec.FromBackup)
 			if err != nil {
-				return nil, werror.NewInvalidError(fmt.Sprintf("cannot get backup and volume name from backup URL %v: %v", volume.Spec.FromBackup, err), "")
+				return nil, werror.NewInternalError(fmt.Sprintf("cannot get backup and volume name from backup URL %v: %v", volume.Spec.FromBackup, err))
 			}
-			bs, err := v.ds.GetBackupRO(bvName)
+			bs, err := v.ds.GetBackupRO(bName)
 			if err != nil {
-				return nil, errors.Wrapf(err, "failed to list backups for volumes: %v", bvName)
+				return nil, werror.NewInternalError(fmt.Sprintf("failed to list backups for volumes: %v", bName))
 			}
 
 			if bs != nil {
 				if labelAccessMode, isExist := bs.Status.Labels[types.GetLonghornLabelKey(types.LonghornLabelVolumeAccessMode)]; isExist {
 					accessModeFromBackup = longhorn.AccessMode(labelAccessMode)
 				}
 			}
-			log.Infof("[volumeMutator] patchOps: %v", fmt.Sprintf(`{"op": "replace", "path": "/spec/accessMode", "value": "%s"}`, string(accessModeFromBackup)))
-			patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/accessMode", "value": "%s"}`, string(accessModeFromBackup)))
-		} else {
-			log.Infof("[volumeMutator] patchOps: %v", fmt.Sprintf(`{"op": "replace", "path": "/spec/accessMode", "value": "%s"}`, string(accessModeFromBackup)))
-			patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/accessMode", "value": "%s"}`, string(accessModeFromBackup)))
 		}
+		patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/accessMode", "value": "%s"}`, string(accessModeFromBackup)))
 	}
```

Comment:
Suggest moveing the change to the previous commit.

Reply:
Merge the commits to avoid make confused.